Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp

```diff
@@ -820,96 +820,134 @@ static Color parseHSLParametersRaw(CSSParserTokenRange& range, ColorParserState&
     return parseNonRelativeHSLParametersRaw(args, state);
 }
 
+// MARK: - hwb()
+
+using ParsedHWBA = std::tuple<AngleOrNumberOrNoneRaw, NumberOrPercentOrNoneRaw, NumberOrPercentOrNoneRaw, double>;
+
+static HWBA<float> normalizeHWBParametersRaw(ParsedHWBA parsedHWBA)
+{
+    auto [hue, whiteness, blackness, alpha] = parsedHWBA;
+
+    auto normalizedHue = WTF::switchOn(hue,
+        [] (AngleRaw angle) { return normalizeHue(CSSPrimitiveValue::computeDegrees(angle.type, angle.value)); },
+        [] (NumberRaw number) { return normalizeHue(number.value); },
+        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
+    );
+    auto normalizedWhiteness = WTF::switchOn(whiteness,
+        [] (PercentRaw percent) { return percent.value; },
+        [] (NumberRaw number) { return number.value; },
+        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
+    );
+    auto normalizedBlackness = WTF::switchOn(blackness,
+        [] (PercentRaw percent) { return percent.value; },
+        [] (NumberRaw number) { return number.value; },
+        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
+    );
+
+    return HWBA<float> {
+        static_cast<float>(normalizedHue),
+        static_cast<float>(normalizedWhiteness),
+        static_cast<float>(normalizedBlackness),
```

Comment:
We could just use `float` rather than `auto` above for these local variables, and avoid the need for these 3 casts.